test/functional/editions_controller_test.rb

```diff
@@ -200,4 +200,59 @@ class EditionsControllerTest < ActionController::TestCase
       end
     end
   end
+
+  context "#update" do
+    setup do
+      @answer = FactoryBot.create(:answer_edition)
+    end
+
+    should "show update and success message and render show template when saved" do
+      post :update, params: {
+        id: @answer.id,
+        edition: {
+          title: "The changed title",
+        },
+      }
+
+      assert_template "show"
+      assert_equal "Edition updated successfully.", flash[:success]
+      assert_equal "The changed title", AnswerEdition.last.title
+    end
+
+    should "show error message and render show template when title field is blank" do
+      post :update, params: {
+        id: @answer.id,
+        edition: {
+          title: "",
+        },
+      }
+
+      assert_template "show"
+      assert_select ".gem-c-error-summary__list-item", "Enter a title"
+    end
+
+    should "show error message and render show template when the edition could not be updated" do
```

Comment:
I think we're missing a test for when `save` returns `false`.

Reply:
I've added a test which might (or might not) do what we'd expect: If we stub save to return false without triggering a validation error we get the standard error rather than a validation or other message. This seems to make sense but if we add a specific message in the controller we don't see this, which would be my expectation. 🤔